transformations/casual2formal/transformation.py

```diff
@@ -0,0 +1,114 @@
+        top_adequacy_phrases = []
+        for para_phrase in para_phrases:
+            x = self.tokenizer.encode(
+                input_phrase,
+                para_phrase,
+                return_tensors="pt",
+                truncation_strategy="only_first",
+            )
+            self.nli_model = self.nli_model.to(device)
+            logits = self.nli_model(x.to(device))[0]
+            # we throw away "neutral" (dim 1) and take the probability of "entailment" (2) as the adequacy score
+            entail_contradiction_logits = logits[:, [0, 2]]
+            probs = entail_contradiction_logits.softmax(dim=1)
+            prob_label_is_true = probs[:, 1]
+            adequacy_score = prob_label_is_true[0].item()
+            if adequacy_score >= adequacy_threshold:
+                top_adequacy_phrases.append(para_phrase)
+        return top_adequacy_phrases
+
+    def score(self, input_phrase, para_phrases, adequacy_threshold, device="cpu"):
+        adequacy_scores = {}
+        for para_phrase in para_phrases:
+            x = self.tokenizer.encode(
+                input_phrase,
+                para_phrase,
+                return_tensors="pt",
+                truncation_strategy="only_first",
+            )
+            self.nli_model = self.nli_model.to(device)
+            logits = self.nli_model(x.to(device))[0]
+            # we throw away "neutral" (dim 1) and take the probability of "entailment" (2) as the adequacy score
+            entail_contradiction_logits = logits[:, [0, 2]]
+            probs = entail_contradiction_logits.softmax(dim=1)
+            prob_label_is_true = probs[:, 1]
+            adequacy_score = prob_label_is_true[0].item()
+            if adequacy_score >= adequacy_threshold:
+                adequacy_scores[para_phrase] = adequacy_score
+        return adequacy_scores
+
+
+class Casual2Formal(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+
+    def __init__(self, num_beams=5, max_length=32, quality_filter=0.95, device=None):
+        super().__init__()
+        if self.verbose:
+            print("Starting to load Casual to Formal Model...\n")
+        m_name = "prithivida/informal_to_formal_styletransfer"
+        self.tokenizer = AutoTokenizer.from_pretrained(m_name)
+        self.model = AutoModelForSeq2SeqLM.from_pretrained(m_name)
+        if self.verbose:
+            print("Completed loading Casual to Formal Model.\n")
+        self.adequacy = Adequacy()
+        self.max_output = num_beams
+        self.num_beams = num_beams
+        self.max_length = max_length
+        self.quality_filter = quality_filter
+        self.device = device
+        if self.device is None:
+            self.device = "cuda" if torch.cuda.is_available() else "cpu"
+
+    def generate(self, sentence: str):
+        ctf_prefix = "transfer Casual to Formal: "
```

Comment:
I think this all seems good now. I noticed that "Adequacy" is present in both the classes and hence would take up twice the memory. I would suggest one refactoring since most of the code in formal2casual and casual2formal is the same. You could make just one transformation folder - maybe called "formality_change", "style_change" or something better and write both the classes inside one single file _transformation.py_. And you could additionally include the active/passive conversion too if that works well.

Reply:
Unlike other transformations, it is okay in your case to load Adequacy (i.e. the model and the tokenizer) outside init (so that it is loaded only once for all 4 models).

Reply:
Combined the two in #152.